spec/p2p/p2p.qnt

```diff
@@ -0,0 +1,577 @@
+
+    // val invConnectionsOpen = modeledNodes.forall(y => config.get(y).activePeers().forall(x => 
+    //     config.get(y).peerState.get(x) == "outbound" or config.get(y).peerState.get(x) == "inbound"))
+
+
+
+/* ****************************************************************************
+ * Actions
+ * How the above functions can be invoked (e.g. by which node)
+ * ************************************************************************* */
+
+  
+/* 
+ * Future syntactic sugar.
+ */
+    // helps for better readability. 
+    def require(pred:bool):bool = pred
+
+/*
+ * Useful actions to simplify spec
+ */ 
+
+    action allUnchanged = all {
+        config' = config,
+        Environment' = Environment,
+        OperatorAddressBook' = OperatorAddressBook,
+        requestingNodes' = requestingNodes,
+    }    
+
+    // Framerule
+    action environmentUnchanged = all {
+        Environment' = Environment,
+        OperatorAddressBook' = OperatorAddressBook,
+        requestingNodes' = requestingNodes,
+    }    
+
+    
+
+/*
+ * Local actions
+ */
+
+   // node dials a. This starts lower lever computations. 
+    action actionDialPeer (node: Address, a:Address):bool = all {
+        config' = config.set(node, config.get(node).dialPeer(a)),
+        environmentUnchanged,
+    }
+
+    action actionTransportAccept (node: Address, a:Address):bool = all {
+        config' = config.set(node, config.get(node).transportAccept(a)),
+        environmentUnchanged,
+    }
+
+    action actionOnLostConnection(node: Address, c: Connection): bool = all {
+        config' = config.set(node, config.get(node).onLostConnection(c)),
+        environmentUnchanged,
+    }
+
+/* 
+ * Environment actions: From the lower level (modeling the outcome of TCP connection
+ * initialization, establishment, etc.)
+ */
+
+    // auxiliary functions
+    pure def removeConnectionOutgoing(s: LocalState, a: Address): LocalState = {
+        val y : Set[Connection] = s.connections.exclude(Set({address: a, state: "outgoing"}))
+        s.with("connections", y)
+    }
+
+    pure def putPeer(s: LocalState, a: Address, direction: str): LocalState = {
+        val y =  s.peerState.put(a, direction)
+        s.with("peerState", y)
+    }
+
+    pure def loseConnection(s: LocalState, a: Address): LocalState = {
+        val y : Set[Connection] = 
+            s.connections.exclude(Set({address: a, state: "outgoing"}, {address: a, state: "incoming"}))
+                         .union(Set({address: a, state: "lost"}))
+        s.with("connections", y)
+    }
+
+ // return DialPeerSuccess -> if connection.address.id not in Peers -> peerState(address) = "outbound" 
+ //                           otherwise -> delete connection from map 
+ // in the spec right now: the peer might not exists, be already outbound (stutter), be reconnecting
+ // TODO: check the cases
+    action lowerLevelDialPeerSuccess(node: Address, a: Address): bool = all {
```

Comment:
This method, in my opinion, should create an inbound connection at the destination peer. So, it would be something like the `transportAccept` at the destination peer.